src/framework/xr/xr-view.js

```diff
@@ -314,17 +518,48 @@ class XrView {
     _onDeviceLost() {
         this._frameBufferSource = null;
         this._frameBuffer = null;
+        this._depthInfo = null;
+    }
+
+    /**
+     * Get depth value from depth information in meters. UV is in range of 0..1, with origin in
+     * top-left corner of a texture.
+     *
+     * @param {number} u - U coordinate of pixel in depth texture, which is in range from 0.0 to
+     * 1.0 (left to right).
+     * @param {number} v - V coordinate of pixel in depth texture, which is in range from 0.0 to
+     * 1.0 (top to bottom).
+     * @returns {number|null} Depth in meters or null if depth information is currently not
+     * available.
+     * @example
+     * const depth = view.getDepth(u, v);
+     * if (depth !== null) {
+     *     // depth in meters
+     * }
+     */
+    getDepth(u, v) {
+        if (this._manager.views.depthGpuOptimized)
+            return null;
+
+        return this._depthInfo?.getDepthInMeters(u, v) ?? null;
     }
 
     /**
      * @ignore
      */
     destroy() {
+        this._depthInfo = null;
+
         if (this._textureColor) {
             this._textureColor.destroy();
             this._textureColor = null;
         }
 
+        if (this._textureDepth) {
+            this._textureDepth.destroy();
```

Comment:
So we override `this._textureDepth.impl._glTexture` and then destroy it.
1. Was _glTexture already allocated before? I suspect not, so we're not leaking it.
2. can we destroy _glTexture provided by XR?

Reply:
We don't allocate it indeed, but for the safety, when the view is destroyed (end of the session), I do destroy that texture also. Subsequent sessions work well, so it seems to be good.